lldb/packages/Python/lldbsuite/test/make/Makefile.rules

```diff
@@ -355,6 +355,16 @@ ifeq "$(OS)" "Windows_NT"
 	endif
 endif
 
+#----------------------------------------------------------------------
+# Darwin cross compilation
+#----------------------------------------------------------------------
+ifeq "$(HOST_OS)" "Darwin"
+        ifneq "$(HOST_OS)" "$(OS)"
+                LDFLAGS += -fuse-ld=lld
```

Comment:
This only works if you have lld built or installed on your machine. I wonder if there's something we can do to warn you about this before all the tests fail to link?

Reply:
I guess we could try to cross compile a simple binary upfront and complain when it fails, but I'd say it's reasonable to just say that you need lld if you want to cross compile. I believe the same thing is true on Windows.

Reply:
I think the failure mode is that you try to run tests (on a mac, targeting a non-mac), we will try to link the test binaries with lld, which will fail immediately.  Someone who doesn't know would (I think) say "Oh, I guess I need lld to build binaries for this remote target" when every test fails the same way.

Is this going to work if I run the lldb testsuite on a macOS system but running the tests on iOS or other Darwin type OS.  Looking through existing OS comparisons in Makefile.rules, it seems like it can be Darwin/Android/Windows_NT/FreeBSD/NetBSD (plus I'm sure Linux).  So probably it calls an iOS device Darwin too.

Reply:
Yeah good point, I was trying to be generic but we shouldn't use lld when targeting a remote Darwin device.